File: bartcode-service/src/main/java/uk/co/bartcode/service/config/RepositoryRestConfig.java
package uk.co.bartcode.service.config;

import org.springframework.context.annotation.Configuration;
import org.springframework.data.rest.core.config.RepositoryRestConfiguration;
import org.springframework.data.rest.webmvc.config.RepositoryRestConfigurerAdapter;
import uk.co.bartcode.service.post.DatePostCount;
import uk.co.bartcode.service.post.Post;
import uk.co.bartcode.service.post.TagPostCount;

@Configuration
class RepositoryRestConfig extends RepositoryRestConfigurerAdapter {

    @Override
    public void configureRepositoryRestConfiguration(RepositoryRestConfiguration config) {
        config.exposeIdsFor(DatePostCount.class, TagPostCount.class, Post.class);
        config.getCorsRegistry().addMapping("/**")
                .allowedOrigins("http://localhost:4000", "http://localhost:4200")
                .allowedMethods("GET");
    }


}

File: bartcode-service/src/main/kotlin/uk/co/bartcode/service/config/RepositoryRestConfig.kt
package uk.co.bartcode.service.config

import org.springframework.context.annotation.Configuration
import org.springframework.data.rest.core.config.RepositoryRestConfiguration
import org.springframework.data.rest.webmvc.config.RepositoryRestConfigurerAdapter
import uk.co.bartcode.service.post.DatePostCount
import uk.co.bartcode.service.post.TagPostCount

@Configuration
class RepositoryRestConfig : RepositoryRestConfigurerAdapter() {

    override fun configureRepositoryRestConfiguration(config: RepositoryRestConfiguration) {
        config.exposeIdsFor(DatePostCount::class.java, TagPostCount::class.java)
        config.corsRegistry.addMapping("/**")
                .allowedOrigins("http://localhost:4000", "http://localhost:4200")
    }

}